app/scripts/controllers/journalUpload.js

```diff
@@ -0,0 +1,241 @@
+          newParameters.page === oldParameters.page
+        ) {
+          $scope.queryParameters.page = 1;
+          return;
+        }
+
+        if (newParameters.page !== oldParameters.page) {
+          //scroll to top on page change
+          $window.scrollTo(0, 0);
+        }
+
+        $scope.refresh();
+      },
+      true,
+    );
+
+    $scope.$watch('currentReportId', (newReportId, oldReportId) => {
+      // scroll to top of page if selected report changes
+      if (newReportId !== oldReportId) {
+        $window.scrollTo(0, 0);
+      }
+      // if currentReportId is null, refresh account transfer data
+      if (!newReportId) {
+        $scope.refresh();
+      } else {
+        // set accountTransfers to current report's valid account transfers
+        $scope.accountTransfers = $scope.reports
+          .filter(report => report.id === newReportId)[0]
+          .accountTransfers.filter(accountTransfer => !accountTransfer.error);
+        // set accountTransfersWithErrors to current report's account transfers that have errors
+        $scope.accountTransfersWithErrors = $scope.reports
+          .filter(report => report.id === newReportId)[0]
+          .accountTransfers.filter(accountTransfer => accountTransfer.error);
+      }
+    });
+
+    const throttleFilter = _.debounce(() => {
+      $scope.$apply(() => {
+        $scope.queryParameters.filter = $scope.strFilter;
+      });
+    }, 500);
+
+    $scope.$watch('strFilter', strFilter => {
+      if (angular.isDefined(strFilter)) {
+        throttleFilter();
+      }
+    });
+
+    $scope.resetStrFilter = () => {
+      $scope.strFilter = '';
+    };
+
+    $scope.refresh = () => {
+      journalUploadService
+        .getRegistrationData(conference.id, $scope.queryParameters)
+        .then(data => {
+          $scope.meta = data.meta;
+          $scope.accountTransfers = journalUploadService.getAccountTransferData(
+            data,
+          );
+          $scope.accountTransfersWithErrors = journalUploadService.getAccountTransferDataWithErrors(
+            data,
+          );
+          $scope.removeAllTransfersFromToInclude();
+        });
+    };
+
+    $scope.refreshAccountTransfersToInclude = accountTransfer => {
+      const accountTransferIndex = $scope.accountTransfersToInclude.indexOf(
+        accountTransfer,
+      );
+      accountTransferIndex !== -1
+        ? $scope.accountTransfersToInclude.splice(accountTransferIndex, 1)
+        : $scope.accountTransfersToInclude.push(accountTransfer);
+    };
+
+    $scope.allAccountTransfersIncluded = () =>
+      $scope.accountTransfersToInclude.length ===
+      $scope.accountTransfers.length;
+
+    $scope.addAllTransfersToInclude = () => {
+      angular.forEach($scope.accountTransfers, accountTransfer => {
+        if ($scope.accountTransfersToInclude.indexOf(accountTransfer) === -1) {
+          accountTransfer.included = true;
+          $scope.accountTransfersToInclude.push(accountTransfer);
```

Comment:
Little odd that you're maintaining this state in two places.

Reply:
It is a bit, once they uncheck to include the account transfer it's removed from ```accountTransfersToInclude``` and the ```included``` value for that account transfer is set back to ```false```. So really the included state only matters in ```$scope.accountTransfers```. Not too sure how to improve it, as it's needed for the checked/unchecked state of the checkbox element.